WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewModel.swift

```diff
@@ -0,0 +1,164 @@
+import Foundation
+import CoreData
+import UIKit
+
+/// Responsible for populating a table view with posts
+///
+class PostsCardViewModel: NSObject {
+    var blog: Blog
+
+    private let managedObjectContext: NSManagedObjectContext
+
+    private let tableView: UITableView
+
+    lazy var filterSettings: PostListFilterSettings = {
+        PostListFilterSettings(blog: blog, postType: .post)
+    }()
+
+    private var fetchedResultsController: NSFetchedResultsController<Post>!
+
+    init(tableView: UITableView, blog: Blog, managedObjectContext: NSManagedObjectContext = ContextManager.shared.mainContext) {
+        self.blog = blog
+        self.managedObjectContext = managedObjectContext
+        self.tableView = tableView
+        super.init()
+    }
+
+
+    /// Refresh the results and reload the data on the table view
+    func refresh() {
+        do {
+            try fetchedResultsController.performFetch()
+            tableView.reloadData()
+        } catch {
+            print("Fetch failed")
+        }
+    }
+
+    /// Set up the view model to be ready for use
+    func viewDidLoad() {
+        createFetchedResultsController()
+    }
+}
+
+// MARK: - Private methods
+
+private extension PostsCardViewModel {
+    func createFetchedResultsController() {
+        // 0 = published, 1 = draft, 2 = scheduled
+        filterSettings.setCurrentFilterIndex(1)
+
+        fetchedResultsController?.delegate = nil
+        fetchedResultsController = nil
+
+        fetchedResultsController = NSFetchedResultsController(fetchRequest: fetchRequest(), managedObjectContext: managedObjectContext, sectionNameKeyPath: nil, cacheName: nil)
+
+        fetchedResultsController.delegate = self
+    }
+
+    func fetchRequest() -> NSFetchRequest<Post> {
+        let fetchRequest = NSFetchRequest<Post>(entityName: String(describing: Post.self))
+        fetchRequest.predicate = predicateForFetchRequest()
+        fetchRequest.sortDescriptors = sortDescriptorsForFetchRequest()
+        fetchRequest.fetchBatchSize = 3
+        fetchRequest.fetchLimit = 3
```

Comment:
What do you think of defining a constant for the batch size / fetch limit?

Reply:
Done 26760b16bbdf4470da167af91d2d765095aa7b23